config-model/src/main/java/com/yahoo/schema/processing/FieldSetSettings.java

```diff
@@ -104,4 +108,21 @@ private void checkStemming(Schema schema, FieldSet fieldSet) {
         }
     }
 
+    private void checkTypes(Schema schema, FieldSet fieldSet) {
+        var tensorFields = new LinkedList<String>();
+        var nonTensorFields = new LinkedList<String>();
+        for (String fieldName : fieldSet.getFieldNames()) {
+            ImmutableSDField field = schema.getField(fieldName);
+            if (field.getDataType() instanceof TensorDataType) {
+                tensorFields.add(field.getName());
+            } else {
+                nonTensorFields.add(field.getName());
+            }
+        }
+        if (!tensorFields.isEmpty() && !nonTensorFields.isEmpty()) {
+            throw new IllegalArgumentException("For schema '" + schema.getName() + "', fieldset '" + fieldSet.getName() + "': " +
+                    "Illegal mixing of tensor fields ['" + String.join("','", tensorFields) + "'] " +
+                    "and non-tensor fields ['" + String.join("','", nonTensorFields) + "']");
```

Comment:
Consider instead something like `"For schema 'X', fieldset 'Y': tensor fields ['foo', 'bar'] cannot be mixed with non-tensor fields ['baz', 'zoid'] in the same fieldset. See https://docs.vespa.ai/en/reference/schema-reference.html#fieldset"`